cmake/buildtype.cmake

```diff
@@ -0,0 +1,10 @@
+##! Set a default build type if none was specified
+set(default_build_type "Release")
+
+if (NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+    message(STATUS "Setting build type to '${default_build_type}' as none was specified.")
+    set(CMAKE_BUILD_TYPE "${default_build_type}" CACHE
+            STRING "Choose the build type." FORCE)
+    # Set the possible values of build type for cmake-gui
+    set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
```

Comment:
MinSizeRel and RelWithDebInfo can be removed since they are Windows build type

Reply:
The names are indeed taken from Windows build types, but I think it might be nice to keep them and compile respectively with flags such as `-Os` and `-O3 -g3`, etc